src/test/db.rs

```diff
@@ -362,3 +370,24 @@ async fn index_option_defaults_test(defaults: Option<IndexOptionDefaults>, name:
     };
     assert_eq!(event_defaults, defaults);
 }
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn watch_test() {
+    let _guard = LOCK.run_concurrently().await;
+
+    let client = EventClient::new().await;
+    if client.is_standalone() {
+        return;
+    }
+    let db = client.database("test_db");
+    let pipeline = vec![doc! { "$match": { "x": 3 }}];
+    let options = ChangeStreamOptions::builder()
+        .selection_criteria(SelectionCriteria::ReadPreference(ReadPreference::Primary))
+        .max_await_time(Duration::from_millis(5))
+        .batch_size(10)
+        .full_document(FullDocumentType::UpdateLookup)
+        .build();
+    let result = db.watch(pipeline, Some(options)).await;
+    assert!(result.is_ok());
```

Comment:
Ditto above about unwrapping instead of asserting

Reply:
fixed